### PR TITLE
sql: add support for intervals, DATE_SUB and DATE_ADD

### LINKED ISSUE #663: DATE_SUB and DATE_ADD functions
See https://github.com/src-d/gitbase/issues/784

### PR DESCRIPTION
**Pull request by erizocosmico**
Fixes #663

This PR introduces a few new features:
- Correctly parsing interval expressions.
- Allow using the + and - operators to subtract from and add to
  dates. e.g. `'2019-04-10' - INTERVAL 1 DAY`.
- New `DATE_ADD` function, which is essentially the same as
  `DATE + INTERVAL`.
- New `DATE_SUB` function, which is essentially the same as
  `DATE - INTERVAL`.
- Validation rule to ensure intervals are only used in certain
  specific places, such as DATE_SUB, DATE_ADD, + and -. Using it
  anywhere else is not valid SQL, but vitess does not catch those
  errors. Plus, even if it's an expression for convenience, its
  `Eval` method is a stub and panics, so it should not be used
  unless it's in an expression that knows how to deal with intervals.

### CODE REVIEW

**Comment by mcuadros** [comment]
I confirm that works properly with superset

**Comment by erizocosmico** [comment]
Rebased

**Comment by erizocosmico** [comment]
Rebased again

File: README.md [hunk 1]
```diff
@@ -70,7 +70,8 @@ We support and actively test against certain third-party clients to ensure compa
 |`COUNT(expr)`| Returns a count of the number of non-NULL values of expr in the rows retrieved by a SELECT statement.|
 |`DATE_ADD(date, interval)`|Adds the interval to the given date.|
 |`DATE_SUB(date, interval)`|Subtracts the interval from the given date.|
-|`DAY(date)`|Returns the day of the given date.|
+|`DAY(date)`|Synonym for DAYOFMONTH().|
+|`DAYOFMONTH(date)`|Return the day of the month (0-31).|
```
Comment (kuba--) [inline README.md]: And I suppose we also need to add it to SUPPORT.md

**Comment by erizocosmico** [comment]
Shall we merge this?

**Comment by ajnavarro** [comment]
@erizocosmico I think changes on SUPPORTED.md file are missed.

**Comment by erizocosmico** [comment]
@ajnavarro the one for DAYOFMONTH? That's not from this PR. DATE_ADD and DATE_SUB are already on SUPPORTED.md

File: README.md (hunk 1 above)
Comment (erizocosmico) [reply README.md]: This is not from this PR

**Pull request merged**